.circleci/config.yml

```diff
@@ -72,20 +73,25 @@ aliases:
         sudo chown -R 1000 ../os-autoinst/t/data
       else # only in local run
          [ -d ../os-autoinst ] || { sudo mkdir ../os-autoinst && sudo chown -R 1000 ../os-autoinst; }
-         bash .circleci/build_autoinst.sh ../os-autoinst
+         bash .circleci/build_autoinst.sh ../os-autoinst $(cat .circleci/autoinst.sha)
        fi
 
+  - &make_test
+    command: |
+      case "$CIRCLE_JOB" in
+      t)         make coverage-codecov CHECKSTYLE=1 PROVE_ARGS="$HARNESS -v t/*.t"     GLOBIGNORE="$(cat .circleci/unstable_tests.txt | tr '\n' :)";;
+      ui)        make coverage-codecov CHECKSTYLE=0 PROVE_ARGS="$HARNESS -v t/ui/*.t"  GLOBIGNORE="$(cat .circleci/unstable_tests.txt | tr '\n' :)";;
+      api)       make coverage-codecov CHECKSTYLE=0 PROVE_ARGS="$HARNESS -v t/api/*.t" GLOBIGNORE="$(cat .circleci/unstable_tests.txt | tr '\n' :)";;
+      unstable)  for f in $(cat .circleci/unstable_tests.txt); do make coverage CHECKSTYLE=0 PROVE_ARGS="$HARNESS -v $f" RETRY=3; done; make coverage-report-codecov;;
+      fullstack) make coverage-codecov CHECKSTYLE=0 FULLSTACK=1           PROVE_ARGS="$HARNESS -v t/full-stack.t" RETRY=3;;
+      scheduler) make coverage-codecov CHECKSTYLE=0 SCHEDULER_FULLSTACK=1 PROVE_ARGS="$HARNESS -v t/05-scheduler-full.t";;
+      developer) make coverage-codecov CHECKSTYLE=0 DEVELOPER_FULLSTACK=1 PROVE_ARGS="$HARNESS -v t/33-developer_mode.t";;
+      build.docs) COMMIT_AUTHOR_EMAIL=skynet@open.qa GEM_HOME=$(pwd)/.gem script/generate-documentation $encrypted_e2c381aa6b8c_key $encrypted_e2c381aa6b8c_iv;;
```

Comment:
Not sure if that's "good form" but I love that each test becomes one line, rather than 20 or so

Reply:
still `make coverage-${CIRCLECI_JOB}` would be better here, shame that you didn't support the PR with Makefile changes

Reply:
I thought you were going to propose this separately because the other one got a bit too complex. Maybe we should discuss it a bit more afterall.